atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java

```diff
@@ -205,6 +205,8 @@ protected boolean shouldStopRetrying(int numTimesFailed) {
                     responses.stream()
                             .map(response -> response.immutableTimestamp().getLock())
                             .collect(Collectors.toSet()));
+            responses.forEach(response -> lockWatchEventCache.removeTransactionStateFromCache(
```

Comment:
So this is fine, only because of the way we wire things up: the timelockService above is TimeLockClient, and in there the tryUnlock is non-blocking and does not throw. Ideally we would not rely on that, but I think it's fine here.

Reply:
Discussed offline